filebeat/input/filestream/config_test.go

```diff
@@ -30,3 +36,158 @@ func TestConfigValidate(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestValidateInputIDs(t *testing.T) {
+	tcs := []struct {
+		name       string
+		cfg        []string
+		assertErr  func(t *testing.T, err error)
+		assertLogs func(t *testing.T, buff *observer.ObservedLogs)
+	}{
+		{
+			name: "empty config",
+			cfg:  []string{""},
+			assertErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, "empty config should not return an error")
+			},
+		},
+		{
+			name: "empty ID",
+			cfg: []string{`
+type: filestream
+`, `
+type: filestream
+id: some-id-1
+`, `
+type: filestream
+id: some-id-2
+`,
+			},
+			assertErr: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "input without ID")
+			},
+			assertLogs: func(t *testing.T, obs *observer.ObservedLogs) {
+				want := `[{"type":"filestream"}]`
+
+				logs := obs.TakeAll()
+				require.Len(t, logs, 1, "there should be only one log entry")
+
+				got, err := json.Marshal(logs[0].ContextMap()["inputs"])
+				require.NoError(t, err, "could not marshal duplicated IDs inputs")
+				assert.Equal(t, want, string(got))
+
+			},
+		},
+		{
+			name: "duplicated IDs",
```

Comment:
We need to add a test case when you have more than 2 duplicates (5?) and the error message does not repeat the same input more than once.